sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -1166,15 +1166,15 @@ def test_gradient_boosting_early_stopping():
     X, y = make_classification(n_samples=1000, random_state=0)
 
     gbc = GradientBoostingClassifier(
-        n_estimators=1000,
+        n_estimators=100,
```

Comment:
The value of `n_estimators` when early stopping is enabled with `n_iter_no_change` should have no impact because the learning algorithm stops much before reaching 100 or 1000 trees. Later in the test you see that only 28 or 70 trees are actually learned for gbc depending on the learning rate:

```python
        (gbc, 1e-1, 28),
        (gbr, 1e-1, 13),
        (gbc, 1e-3, 70),
        (gbr, 1e-3, 28),
```

Let's revert this change:

```suggestion
        n_estimators=1000,
```